3Dmol/glmodel.js

```diff
@@ -51,57 +51,209 @@ $3Dmol.GLModel = (function() {
         "NI" : 1.63,
         "Ni" : 1.63
     };
+    /*
+        The dictioanries are for dropdown menus and validation of the viewer
+    */
+
+    var validElements = [
+        "H",
+        "Li",
+        "LI",
+        "Na",
+        "NA",
+        "K",
+        "C",
+        "N",
+        "O",
+        "F",
+        "P",
+        "S",
+        "CL",
+        "Cl",
+        "BR",
+        "Br",
+        "SE",
+        "Se",
+        "ZN",
+        "Zn",
+        "CU",
+        "Cu",
+        "NI",
+        "Ni"
+    ]
+
+    var validColorSpecs=[
+        "white",
+        "silver",
+        "gray",
+        "grey",
+        "black",
+        "red",
+        "maroon",
+        "yellow",
+        "orange",
+        "olive",
+        "lime",
+        "green",
+        "aqua",
+        "cyan",
+        "teal",
+        "blue",
+        "navy",
+        "fuchsia",
+        "magenta",
+        "purple",
+    ]
+
+    var validColorSchemeSpecs =[
+        "greenCarbon",
+        "cyanCarbon",
+        "magentaCarbon",
+        "yellowCarbon",
+        "whiteCarbon",
+        "orangeCarbon",
+        "purpleCarbon",
+        "blueCarbon",
+        "ssPyMOL",
+        "ssJmol",
+        "Jmol",
+        "default",
+        "amino",
+        "shapely",
+        "nucleic",
+        "chain",
+        "chainHetatm",
+    ]
+
+    var validAtomSpecs = {
+        "resn":{type:"string",valid :true}, // Parent residue name
```

Comment:
I don't like "valid".  There's nothing wrong with it, just not appropriate for the viewer.  How about "gui"